YoCode/CommandLineArguments/CommandLineParser.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+
+
+namespace YoCode
+{
+    public class CommandLineParser
+    {
+        List<string> implementedCommands = new List<string>() { CommandNames.ORIGIN, CommandNames.MODIFIED, CommandNames.HELP };
+        List<SplitArg> currentCommands;
+        
+        public CommandLineParser(string[] args)
+        {
+            currentCommands = args.Select(arg => CommandExtractor.ArgsSplitter(arg)).ToList();
+        }
+
+        public InputResult Parse()
+        {
+            var ires = new InputResult();
+
+            foreach (SplitArg arg in currentCommands)
+            {
+                ires.originalFilePath = (arg.command == CommandNames.ORIGIN) ? arg.data : ires.originalFilePath;
+                ires.modifiedFilePath = (arg.command == CommandNames.MODIFIED) ? arg.data : ires.modifiedFilePath;
+                ires.helpAsked = arg.command == CommandNames.HELP;
```

Comment:
Why are `ires.originalFilePath` and `ires.modifiedFilePath` setting themselves to... themselves... if there's no argument? It doesn't serve any purpose, it just looks odd. Another way to structure this code might be:
```
switch (arg.command)
{
    case CommandNames.ORIGIN:
        ires.originalFilePath = arg.data;
        break;
    case CommandNames.MODIFIED:
        ires.modifiedFilePath = arg.data;
        break;
    case CommandNames.HELP:
        ires.helpAsked = true;
        break;
}
```